numpyro/infer/initialization.py

```diff
@@ -22,13 +23,17 @@ def init_to_median(site=None, num_samples=15):
     if site['type'] == 'sample' and not site['is_observed'] and not site['fn'].is_discrete:
         rng_key = site['kwargs'].get('rng_key')
         sample_shape = site['kwargs'].get('sample_shape')
-        samples = site['fn'].sample(rng_key, sample_shape=(num_samples,) + sample_shape)
-        return np.median(samples, axis=0)
+        try:
+            samples = site['fn'].sample(rng_key, sample_shape=(num_samples,) + sample_shape)
+            return np.median(samples, axis=0)
+        except NotImplementedError:
+            return init_to_uniform(site)
```

Comment:
Can we just return `prototype_value = np.full(site['fn'].shape(), np.nan)` instead of calling `init_to_uniform` so that we don't need to additional work like calling `random.split`?

Reply:
For substitute_fn, I think it only works if we substitue(seed(model), ...) because seed will give rng_key for the site we want to apply substitute_fn. Inside substitute_fn, we dont use sample primitive, so there is no random.split here IIUC.